### PR TITLE
Global settings behavior refactoring

### LINKED ISSUE #16: Add "Minimize on Close" setting to UI
Add "Minimize on Close" setting to UI.
By default it must be **Off** because this behavior confused almost everyone.

### LINKED ISSUE #17: Save settings on change (not on Exit)

### PR DESCRIPTION
**Pull request by GrimMaple**
I refactored the settings class to fix some issues with the program.

Fixes #17 
Fixes #16

### CODE REVIEW

**Comment by denis-adamchuk** [comment]
I've reproduced behavior that caused problems with a naive approach to 'save settings more often'. 
But now it seems that I understand what causes that. I think it will be reproducable after your changes too. Please test.
Steps:
1) Prepare a config with 
ShowPublicOnly=false
2) LastSelectedHost=<a host with 100+ projects>
3) LastSelectedProject=<a random project from all available 100+ project>
The problem is that getAllProjects() reads only 100 projects (due to the fact that I did not know about GitLab pagination when implemented that function) and if LastSelectedProject is not among 100 received projects, updateProjectsDropdownList() will not find it and execute the following snippet:
```
         else if (comboBoxProjects.Items.Count > 0)
         {
            comboBoxProjects.SelectedIndex = 0;
         }
```
Which in turn rewrites a stored LastSelectedProject with the first project in the list (because of saving from 
ComboBoxProjects_SelectedIndexChanged!).
I would like to add pagination support (ie "an ability load 100+ projects in GetAllProjects(bool)") by myself a bit later. This should fix a problem described above. Since ShowPublicOnly is normally 'true' I think we may proceed with your changes if even they are sensitive to this bug.

File: mrHelper/mrHelperForm.cs [hunk 1]
```diff
@@ -583,23 +591,6 @@ private void saveConfiguration()
          {
```
[the remark this answers is not in the document]
Comment (GrimMaple) [reply mrHelper/mrHelperForm.cs]: removed

File: mrHelper/mrHelperForm.cs [hunk 2]
```diff
@@ -105,7 +106,7 @@ private void MrHelperForm_FormClosing(object sender, FormClosingEventArgs e)
       {
          try
          {
-            if (!_exiting)
+            if (checkBoxMinimizeOnClose.Checked && !_exiting)
             {
                onHideToTray(e);
                return;
```
[the remark this answers is not in the document]
Comment (GrimMaple) [reply mrHelper/mrHelperForm.cs]: Removed.

File: mrHelper/mrHelperForm.cs [hunk 3]
```diff
@@ -1053,5 +1051,25 @@ public VersionComboBoxItem(string sha, string text, DateTime? timeStamp)
             TimeStamp = timeStamp;
          }
       }
+
```
[the remark this answers is not in the document]
Comment (GrimMaple) [reply mrHelper/mrHelperForm.cs]: Resolved.

**Pull request merged**